# This is a patch for Astro-FITS-Header-2.6.2 to update it to Astro-FITS-Header-2.6.2.new
# 
# To apply this patch:
# STEP 1: Chdir to the source directory.
# STEP 2: Run the 'applypatch' program with this patch file as input.
#
# If you do not have 'applypatch', it is part of the 'makepatch' package
# that you can fetch from the Comprehensive Perl Archive Network:
# http://www.perl.com/CPAN/authors/Johan_Vromans/makepatch-x.y.tar.gz
# In the above URL, 'x' should be 2 or higher.
#
# To apply this patch without the use of 'applypatch':
# STEP 1: Chdir to the source directory.
# STEP 2: Run the 'patch' program with this file as input.
#
#### End of Preamble ####

#### Patch data follows ####
diff -c 'Astro-FITS-Header-2.6.2/ChangeLog' 'Astro-FITS-Header-2.6.2.new/ChangeLog'
Index: ./ChangeLog
*** ./ChangeLog	Mon Jun 23 16:12:05 2003
--- ./ChangeLog	Mon Jun 23 16:47:36 2003
***************
*** 1,3 ****
--- 1,10 ----
+ 2003-06-23  Diab Jerius  <djerius@cfa.harvard.edu>
+ 
+ 	* Item/Item.pm (configure): automatically sets type to COMMENT
+ 	when keyword is COMMENT, HISTORY, or is blank.
+ 	(_stringify): comment keyword values/comments are in columns 9-80,
+ 	unlike other cards which are in columns 11-80.
+ 
  2003-05-27  Diab Jerius  <djerius@cfa.harvard.edu>
  
  	* t/parse.t: added test for index() regexep functionality.
***************
*** 7,14 ****
  	replacebyname(), and removebyname() now call index()
  	instead of doing the search themselves, so they inherit
  	the new behavior.
! 	  
! 
  2003-04-28  Alasdair Allan <aa@astro.ex.ac.uk> (Version 2.6.2)
  
          * Applied patch by Diab, shipped as 2.6.2
--- 14,20 ----
  	replacebyname(), and removebyname() now call index()
  	instead of doing the search themselves, so they inherit
  	the new behavior.
! 	
  2003-04-28  Alasdair Allan <aa@astro.ex.ac.uk> (Version 2.6.2)
  
          * Applied patch by Diab, shipped as 2.6.2
diff -c 'Astro-FITS-Header-2.6.2/Item/Item.pm' 'Astro-FITS-Header-2.6.2.new/Item/Item.pm'
Index: ./Item/Item.pm
Prereq:  1.16 
*** ./Item/Item.pm	Mon Jun 23 16:11:58 2003
--- ./Item/Item.pm	Mon Jun 23 15:55:26 2003
***************
*** 311,316 ****
--- 311,319 ----
        my $method = lc($key);
        $self->$method( $hash{$key}) if exists $hash{$key};
      }
+     # COMMENT, HISTORY, and blank cards are special
+     $self->type('COMMENT') if $self->keyword =~ /^(COMMENT|HISTORY|)$/;
+ 
      # End cards are special, need only do a Keyword => 'END' to configure
      $self->type('END') if $self->keyword() eq 'END';
    }
***************
*** 615,622 ****
  
    } elsif (defined $type && $type eq 'COMMENT') {
  
!     # Comments are from character 11 - 80
!     $card = sprintf("%-10s%-70s", $card, $comment);
  
    } elsif (!defined $type && !defined $value && !defined $comment) {
  
--- 618,625 ----
  
    } elsif (defined $type && $type eq 'COMMENT') {
  
!     # Comments are from character 9 - 80
!     $card = sprintf("%-8s%-72s", $card, $comment);
  
    } elsif (!defined $type && !defined $value && !defined $comment) {
  
#### End of Patch data ####

#### ApplyPatch data follows ####
# Data version        : 1.0
# Date generated      : Mon Jun 23 16:47:38 2003
# Generated by        : makepatch 2.00
# Recurse directories : Yes
# p 'ChangeLog' 8206 1056401256 0100644
# p 'Item/Item.pm' 17515 1056398126 0100644
#### End of ApplyPatch data ####

#### End of Patch kit [created: Mon Jun 23 16:47:38 2003] ####
#### Checksum: 101 3445 59871 ####
